disperser/batcher/finalizer.go

```diff
@@ -179,7 +179,8 @@ func (f *finalizer) updateBlobs(ctx context.Context, metadatas []*disperser.Blob
 		confirmationBlockNumber, err := f.getTransactionBlockNumber(ctx, confirmationMetadata.ConfirmationInfo.ConfirmationTxnHash)
 		if errors.Is(err, ethereum.NotFound) {
 			// The confirmed block is finalized, but the transaction is not found. It means the transaction should be considered forked/invalid and the blob should be considered as failed.
-			_, err := f.blobStore.HandleBlobFailure(ctx, m, f.maxNumRetriesPerBlob)
+			f.logger.Warn("confirmed transaction not found", "blobKey", blobKey.String(), "confirmationTxnHash", confirmationMetadata.ConfirmationInfo.ConfirmationTxnHash.Hex(), "confirmationBlockNumber", confirmationMetadata.ConfirmationInfo.ConfirmationBlockNumber)
+			err := f.blobStore.MarkBlobFailed(ctx, m.GetBlobKey())
```

Comment:
From conversation with @teddyknox , it seems like marking a blob as failed in case of reorg like this is more appropriate than retrying dispersal

Reply:
I think there are few edge cases to consider here.
1. Reorg unconfirms the blob and is marked as `Failed` before the sequencer retrieves the blob status: The sequencer will see the blob as `Failed` and redisperse the blob. 
2. Reorg unconfirms the blob. The sequencer has already retrieved the confirmed blob status. The sequencer submits inbox transaction.
  a. The inbox transaction gets reorg'd as well. Rollup redisperses the same data. 
  b. The inbox transaction successfully makes it to the chain. Now it doesn't matter if the old blob gets retried or is marked as `Failed`. Need DA cert verification.

I think the finalizer is still necessary to be the source of truth though.